app/parse.py

```diff
@@ -15,8 +47,72 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def normalize_text(text: str) -> str:
+    return text.replace("\xa0", " ").strip()
+
+
+def write_products_to_csv(products: list, output_csv_path: str) -> None:
+    with open(output_csv_path + ".csv", "w", newline="") as f:
+        writer = csv.writer(f)
+        writer.writerow(PRODUCT_FIELDS)
+        writer.writerows([astuple(product) for product in products])
+
+
+def get_soup(page_source: str) -> BeautifulSoup:
+    return BeautifulSoup(page_source, "html.parser")
+
+
+def get_product(product: Tag) -> Product:
+    return Product(
+        title=normalize_text(
+            product.select_one("a.title")["title"]),
```

Comment:
Ensure that `product.select_one("a.title")` returns a valid element before accessing `['title']`. This could raise a `TypeError` if the element is not found. Consider adding a check or handling the potential exception.